pkg/agent/openflow/multicluster.go

```diff
@@ -0,0 +1,189 @@
+			Action().SetTunnelDst(tunnelPeer). // Flow based tunnel. Set tunnel destination.
+			Action().LoadRegMark(ToTunnelRegMark).
+			Action().GotoTable(L3DecTTLTable.GetID()).
+			Done(),
+	)
+	return flows
+}
+
+func (f *featureMulticluster) l3FwdFlowToRemoteGatewayViaTun(
+	localGatewayMAC net.HardwareAddr,
+	peerServiceCIDR net.IPNet,
+	tunnelPeer net.IP,
+	remoteGatewayIP net.IP) []binding.Flow {
+	ipProtocol := getIPProtocol(remoteGatewayIP)
+	cookieID := f.cookieAllocator.Request(f.category).Raw()
+	var flows []binding.Flow
+	flows = append(flows,
+		// This generates the flow to forward cross-cluster request packets based
+		// on Service ClusterIP range.
+		L3ForwardingTable.ofTable.BuildFlow(priorityNormal).
+			Cookie(cookieID).
+			MatchProtocol(ipProtocol).
+			MatchDstIPNet(peerServiceCIDR).
+			Action().SetSrcMAC(localGatewayMAC).                 // Rewrite src MAC to local gateway MAC.
+			Action().SetDstMAC(GlobalVirtualMACForMulticluster). // Rewrite dst MAC to virtual Multi-cluster MAC.
+			Action().SetTunnelDst(tunnelPeer).                   // Flow based tunnel. Set tunnel destination.
+			Action().LoadRegMark(ToTunnelRegMark).
+			Action().LoadInPort(0). // Load the in_port to 0 so packets can be output to the same in_port if needed.
+			Action().GotoTable(L3DecTTLTable.GetID()).
+			Done(),
+		// This generates the flow to forward cross-cluster reply traffic based
+		// on Gateway IP.
+		L3ForwardingTable.ofTable.BuildFlow(priorityNormal).
+			Cookie(cookieID).
+			MatchProtocol(ipProtocol).
+			MatchDstIP(remoteGatewayIP).
+			MatchCTStateRpl(true).
+			MatchCTStateTrk(true).
+			Action().SetSrcMAC(localGatewayMAC).
+			Action().SetDstMAC(GlobalVirtualMACForMulticluster).
+			Action().SetTunnelDst(tunnelPeer). // Flow based tunnel. Set tunnel destination.
+			Action().LoadRegMark(ToTunnelRegMark).
+			Action().LoadInPort(0). // Load the in_port to 0 so packets can be output to the same in_port if needed.
+			Action().GotoTable(L3DecTTLTable.GetID()).
+			Done(),
+	)
+	return flows
+}
+
+func (f *featureMulticluster) tunnelClassifierFlow(category cookie.Category, tunnelOFPort uint32) binding.Flow {
+	return ClassifierTable.ofTable.BuildFlow(priorityHigh).
+		Cookie(f.cookieAllocator.Request(category).Raw()).
+		MatchInPort(tunnelOFPort).
+		MatchDstMAC(GlobalVirtualMACForMulticluster).
+		Action().LoadRegMark(FromTunnelRegMark).
+		Action().LoadRegMark(RewriteMACRegMark).
+		Action().LoadInPort(0). // Load the in_port to 0 so packets can be output to the same in_port if needed.
+		Action().GotoStage(stageConntrackState).
+		Done()
+}
+
+func (f *featureMulticluster) snatConntrackFlows(serviceCIDR net.IPNet, localGatewayIP net.IP) []binding.Flow {
+	cookieID := f.cookieAllocator.Request(f.category).Raw()
+	var flows []binding.Flow
+	ipProtocol := getIPProtocol(localGatewayIP)
+	flows = append(flows,
+		// This generates the flow to transform destination IP of reply packets from tracked
```

Comment:
Do you mean "flow to restore destination IP of reply packets of cross-cluster Service connections committed in the SNAT CT zone"?

Reply:
Fixed.